nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/util/SCPTransfer.java

```diff
@@ -0,0 +1,682 @@
+            sftp.rm(remoteDirectoryName);
+        } catch (final SftpException e) {
+            throw new IOException("Failed to delete remote directory " + remoteDirectoryName, e);
+        }
+    }
+
+    @Override
+    public void ensureDirectoryExists(final FlowFile flowFile, final File directoryName) throws IOException {
+        final ChannelSftp channel = getChannel(flowFile);
+        final String remoteDirectory = directoryName.getAbsolutePath().replace("\\", "/").replaceAll("^.\\:", "");
+
+        // if we disable the directory listing, we just want to blindly perform the mkdir command,
+        // eating any exceptions thrown (like if the directory already exists).
+        if (disableDirectoryListing) {
+            try {
+                channel.mkdir(remoteDirectory);
+            } catch (SftpException e) {
+                if (e.id != ChannelSftp.SSH_FX_FAILURE) {
+                    throw new IOException("Could not blindly create remote directory due to " + e.getMessage(), e);
+                }
+            }
+            return;
+        }
+        // end if disableDirectoryListing
+
+        boolean exists;
+        try {
+            channel.stat(remoteDirectory);
+            exists = true;
+        } catch (final SftpException e) {
+            if (e.id == ChannelSftp.SSH_FX_NO_SUCH_FILE) {
+                // No Such File
+                exists = false;
+            } else {
+                throw new IOException("Failed to determine if remote directory exists at " + remoteDirectory + " due to " + e, e);
+            }
+        }
+
+        if (!exists) {
+            // first ensure parent directories exist before creating this one
+            if (directoryName.getParent() != null && !directoryName.getParentFile().equals(new File(File.separator))) {
+                ensureDirectoryExists(flowFile, directoryName.getParentFile());
+            }
+            logger.debug("Remote Directory {} does not exist; creating it", new Object[] {remoteDirectory});
+            try {
+                channel.mkdir(remoteDirectory);
+                logger.debug("Created {}", new Object[] {remoteDirectory});
+            } catch (final SftpException e) {
+                throw new IOException("Failed to create remote directory " + remoteDirectory + " due to " + e, e);
+            }
+        }
+    }
+
+    private ChannelSftp getChannel(final FlowFile flowFile) throws IOException {
+        if (sftp != null) {
+            String sessionhost = session.getHost();
+            String desthost = ctx.getProperty(HOSTNAME).evaluateAttributeExpressions(flowFile).getValue();
+            if (sessionhost.equals(desthost)) {
+                // destination matches so we can keep our current session
+                return sftp;
+            } else {
+                // this flowFile is going to a different destination, reset session
+                close();
+            }
+        }
+
+        final JSch jsch = new JSch();
```

Comment:
Unfortunately Jsch does not provide a default logger. I mean they do and its ```dev/null```, so when things go down there is nothing to see in the logs other then exception which may not even have an associated message. Consider implementing Logger hooking it up to internal logger or Processor's logger. Something like this:
```
Logger l = LoggerFactory.getLogger(SCPTransferTest.class);
        JSch.setLogger(new Logger() {

            @Override
            public void log(int level, String message) {
                switch (level) {
                case Logger.INFO:
                    l.info(message);
                    break;
                case Logger.DEBUG:
                    l.debug(message);
                    break;
                default:
                    break;
                }
            }

            @Override
            public boolean isEnabled(int level) {
                return true;
            }
        });
```